test/units/module_utils/common/text/test_converters.py

```diff
@@ -0,0 +1,203 @@
+    @pytest.mark.parametrize(
+        'test_input,expected',
+        [
+            (set([1]), [1]),
+            (set([2]), [2]),
+        ]
+    )
+    def test_set(self, test_input, expected):
+        """
+        Test for passing a set object to _json_encode_fallback().
+        """
+
+        assert _json_encode_fallback(test_input) == expected
+
+    @pytest.mark.parametrize(
+        'test_input',
+        [
+            1,
+            1.1,
+            'string',
+            [1, 2],
+            True,
+            None,
+            {1: 1},
+            (1, 2),
+        ]
+    )
+    def test_default_behavior(self, test_input):
+        """
+        Test for _json_encode_fallback() using other types.
+
+        It must fail with TypeError.
+        """
+
+        with pytest.raises(TypeError):
+            _json_encode_fallback(test_input)
+
+
+class TestContainerToBytes:
+
+    """Namespace for testing container_to_bytes()."""
+
+    @pytest.mark.parametrize(
+        'test_input,expected',
+        [
+            ({1: 1}, {1: 1}),
+            ([1, 2], [1, 2]),
+            ((1, 2), (1, 2)),
+            (1, 1),
+            (1.1, 1.1),
+            ('str', b'str'),
+            (['str'], [b'str']),
+            (('str'), (b'str')),
+            ({'str': 'str'}, {b'str': b'str'}),
+            ({u'str': u'str'}, {b'str': b'str'}),
+        ]
+    )
+    @pytest.mark.parametrize('encoding', ['utf-8', 'latin1'])
+    @pytest.mark.parametrize('errors', ['strict', 'ignore'])
+    def test_to_bytes_different_types(self, test_input, expected, encoding, errors):
+        """Test for passing objects to container_to_bytes()."""
+
+        assert container_to_bytes(test_input, encoding=encoding, errors=errors) == expected
+
+    @pytest.mark.parametrize(
+        'test_input,expected',
+        [
+            ({1: 1}, {1: 1}),
+            ([1, 2], [1, 2]),
+            ((1, 2), (1, 2)),
+            (1, 1),
+            (1.1, 1.1),
+            (True, True),
+            (None, None),
+            ('str', 'str'.encode(DEFAULT_ENCODING)),
+            (b'str', 'str'.encode(DEFAULT_ENCODING)),
+            (u'str', 'str'.encode(DEFAULT_ENCODING)),
+            (['str'], ['str'.encode(DEFAULT_ENCODING)]),
+            (('str'), ('str'.encode(DEFAULT_ENCODING))),
+            ({'str': 'str'}, {'str'.encode(DEFAULT_ENCODING): 'str'.encode(DEFAULT_ENCODING)}),
+        ]
+    )
+    def test_to_bytes_default_encoding_err(self, test_input, expected):
+        """
+        Test for passing objects to container_to_bytes(). Default encoding and errors
+        """
+
+        assert container_to_bytes(test_input) == expected
+
+
+class TestContainerToText:
+
+    """Namespace for testing container_to_text()."""
+
+    @pytest.mark.parametrize(
+        'test_input,expected',
+        [
+            ({1: 1}, {1: 1}),
+            ([1, 2], [1, 2]),
+            ((1, 2), (1, 2)),
+            (1, 1),
+            (1.1, 1.1),
+            ('str', 'str'),
```

Comment:
You should probably expect unicode strings

Reply:
I don't really understand what's going on ;) you can add this, I'll merge

Reply:
I figured it out, fixed